tests/rust/nested_nonzero.rs

```diff
@@ -0,0 +1,16 @@
+#[repr(transparent)]
+struct Handle<T> {
+    value: std::num::NonZeroU32,
+    _phantom: std::marker::PhantomData<T>,
+}
+
+struct File;
+
+#[repr(C)]
+struct Node {
+    file: Handle<File>,
+    maybe_file: Option<Handle<File>>,
```

Comment:
One thing I don't like about this patch is that you end up with different types for these two fields.